clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java

```diff
@@ -140,41 +145,78 @@ private void handleMetalakeCommand() {
-    } else if (CommandActions.UPDATE.equals(command)) {
-      if (line.hasOption(GravitinoOptions.COMMENT)) {
-        String comment = line.getOptionValue(GravitinoOptions.COMMENT);
-        newUpdateMetalakeComment(url, ignore, metalake, comment).handle();
-      }
-      if (line.hasOption(GravitinoOptions.RENAME)) {
-        String newName = line.getOptionValue(GravitinoOptions.RENAME);
-        boolean force = line.hasOption(GravitinoOptions.FORCE);
-        newUpdateMetalakeName(url, ignore, force, metalake, newName).handle();
+    Map<Set<String>, Runnable> commandMap = new HashMap<>();
+
+    commandMap.put(
+        Sets.newHashSet(CommandActions.DETAILS),
+        () -> {
+          newMetalakeDetails(url, ignore, metalake).handle();
+        });
+    commandMap.put(
+        Sets.newHashSet(CommandActions.DETAILS, GravitinoOptions.AUDIT),
+        () -> {
+          newMetalakeAudit(url, ignore, metalake).handle();
+        });
+    commandMap.put(
+        Sets.newHashSet(CommandActions.LIST), () -> newListMetalakes(url, ignore).handle());
+    commandMap.put(
+        Sets.newHashSet(CommandActions.CREATE),
+        () -> {
+          String comment = line.getOptionValue(GravitinoOptions.COMMENT);
+          newCreateMetalake(url, ignore, metalake, comment).handle();
+        });
+    commandMap.put(
+        Sets.newHashSet(CommandActions.DELETE),
+        () -> {
+          boolean force = line.hasOption(GravitinoOptions.FORCE);
+          newDeleteMetalake(url, ignore, force, metalake).handle();
+        });
+    commandMap.put(
+        Sets.newHashSet(CommandActions.SET, GravitinoOptions.PROPERTY, GravitinoOptions.VALUE),
+        () -> {
+          String property = line.getOptionValue(GravitinoOptions.PROPERTY);
+          String value = line.getOptionValue(GravitinoOptions.VALUE);
+          newSetMetalakeProperty(url, ignore, metalake, property, value).handle();
+        });
+    commandMap.put(
+        Sets.newHashSet(CommandActions.REMOVE, GravitinoOptions.PROPERTY),
+        () -> {
+          String property = line.getOptionValue(GravitinoOptions.PROPERTY);
+          newRemoveMetalakeProperty(url, ignore, metalake, property).handle();
+        });
+    commandMap.put(
+        Sets.newHashSet(CommandActions.PROPERTIES),
+        () -> newListMetalakeProperties(url, ignore, metalake).handle());
+    commandMap.put(
+        Sets.newHashSet(CommandActions.UPDATE, GravitinoOptions.COMMENT),
+        () -> {
+          String comment = line.getOptionValue(GravitinoOptions.COMMENT);
+          newUpdateMetalakeComment(url, ignore, metalake, comment).handle();
+        });
+    commandMap.put(
+        Sets.newHashSet(CommandActions.UPDATE, GravitinoOptions.RENAME),
+        () -> {
+          String newName = line.getOptionValue(GravitinoOptions.RENAME);
+          boolean force = line.hasOption(GravitinoOptions.FORCE);
+          newUpdateMetalakeName(url, ignore, force, metalake, newName).handle();
+        });
```

Comment:
Looks like every time you have to build a full `commandMap` even for one command, am I right? This is quite heavy, do we have a better solution?

Reply:
Anyway, just tell me if you want to go with this option or not.

Reply:
The new style is hard to read I think, and the performance benefit is limited.

In my opinion, switch-case is better than if-else-if here: the code will be more readable, and the performance is better.

Reply:
BTW, there is a build error need be solved.

Reply:
You can ignore the build error, as I said above. This solution has no performance benefit, but that doesn't matter as a user will not notice a few milliseconds. There is a benefit in that we have exact matching with the set of valid commands, and it gets around some limitations of if/else style coding.

Changing to a switch/case would just be a cosmetic change, but it can be done; again, for this, the performance isn't an issue and would be of negligible difference. Note that in some cases, if/else is faster than a switch, especially if some options are more common than others as they are here.

Reply:
I made a PR to move to a switch/case,it does have a couple of down sides, including variable scope issues.